modules/apps/29-fee/keeper/escrow.go

```diff
@@ -26,64 +26,63 @@ func (k Keeper) EscrowPacketFee(ctx sdk.Context, identifiedFee types.IdentifiedP
 		return sdkerrors.Wrapf(types.ErrRefundAccNotFound, "account with address: %s not found", refundAcc)
 	}
 
-	coins := identifiedFee.Fee.RecvFee
-	coins = coins.Add(identifiedFee.Fee.AckFee...)
-	coins = coins.Add(identifiedFee.Fee.TimeoutFee...)
-
-	if err := k.bankKeeper.SendCoinsFromAccountToModule(
-		ctx, refundAcc, types.ModuleName, coins,
-	); err != nil {
+	coins := identifiedFee.Fee.Total()
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, refundAcc, types.ModuleName, coins); err != nil {
 		return err
 	}
 
-	// Store fee in state for reference later
-	k.SetFeeInEscrow(ctx, identifiedFee)
+	packetFees := []types.IdentifiedPacketFee{identifiedFee}
+	if feesInEscrow, found := k.GetFeesInEscrow(ctx, identifiedFee.PacketId); found {
+		packetFees = append(packetFees, feesInEscrow.PacketFees...)
+	}
+
+	identifiedFees := types.NewIdentifiedPacketFees(packetFees)
+	k.SetFeesInEscrow(ctx, identifiedFee.PacketId, identifiedFees)
```

Comment:
We shouldn't be storing the IdentifiedFee in the value, since packetID is already in the key and we'll have the packetID repeated in each element of the list.

We can use IdentifiedPacketFee in genesis, but let's have a separate type `[]PacketFee` that we set in the store.

Reply:
Totally agree, I've outlined this in the description above as I saw you had created #760. I thought this would be best kept for a follow up to reduce the number of changes in this PR. See the suggested proto definitions above, I think we're on the page!

Reply:
Sounds good!